mvebu:puzzle-m902: add network interface name reorder function

IanChang-IEI opened the pull request:
mvebu: puzzle-m902: add network interface name reorder function

In order for the name of the network port to match the name of the device printed text Original: eth2 eth1 eth5 eth4 eth8 eth7 eth0 eth3 eth6 Reorder : eth0 eth1 eth2 eth3 eth4 eth5 eth10 eth11 eth12

tmn505 commented:
Use aliases node in device dts to achieve same result, instead of using `06_puzzle-M902_reorder_eth`.

BTW. this picture: https://www.ieiworld.com/_attach_file/product/photo/748_1627621648_PUZZLE-M902-Software-Defined-Router-600-4.png 
shows that ports are starting from digit "1", so the proposed changes would assign eth0 to port 1, eth1 to port 2 [...] and eth10 to port 1, eth11 to port 2 [...]. That makes it confusing and inconsistent. Start numbering from eth1 and eth11, unless shipped documentation states otherwise, then disregard my proposal.

IanChang-IEI commented:
@tmn505 
I tried to use alias node but it has no effect, so that's why use 06_puzzle-M902_reorder_eth way

Below is the modification I used for testing
```
--- a/arch/arm64/boot/dts/marvell/cn9132-puzzle-m902.dts
+++ b/arch/arm64/boot/dts/marvell/cn9132-puzzle-m902.dts
@@ -29,15 +29,15 @@
                gpio4 = &cp1_gpio2;
                gpio5 = &cp2_gpio1;
                gpio6 = &cp2_gpio2;
-               ethernet0 = &cp0_eth0;
+               ethernet0 = &cp0_eth2;
                ethernet1 = &cp0_eth1;
-               ethernet2 = &cp0_eth2;
-               ethernet3 = &cp1_eth0;
-               ethernet4 = &cp1_eth1;
-               ethernet5 = &cp1_eth2;
-               ethernet6 = &cp2_eth0;
-               ethernet7 = &cp2_eth1;
-               ethernet8 = &cp2_eth2;
+               ethernet2 = &cp1_eth2;
+               ethernet3 = &cp1_eth1;
+               ethernet4 = &cp2_eth2;
+               ethernet5 = &cp2_eth1;
+               ethernet6 = &cp0_eth0;
+               ethernet7 = &cp1_eth0;
+               ethernet8 = &cp2_eth0;
                spi1 = &cp0_spi0;
                spi2 = &cp0_spi1;
                led-boot = &led_power;
```

IanChang-IEI committed:
mvebu: puzzle-m902: add network interface name reorder function

In order for the name of the network port to match the name of the device printed text
Original: eth2 eth1 eth5 eth4 eth8 eth7 eth0 eth3 eth6
Reorder : eth0 eth1 eth2 eth3 eth4 eth5 eth10 eth11 eth12

Signed-off-by: Ian Chang <ianchang@ieiworld.com>
